src/DvsaLogger/Listener/ApiResponse.php

```diff
@@ -2,19 +2,19 @@
 
 namespace DvsaLogger\Listener;
 
-use Laminas\EventManager\EventInterface;
 use Laminas\EventManager\EventManagerInterface;
 use Laminas\EventManager\ListenerAggregateInterface;
 use Laminas\Log\Logger as Log;
 use Laminas\Log\LoggerAwareInterface;
 use Laminas\Log\LoggerAwareTrait;
 use Laminas\Mvc\MvcEvent;
-use Laminas\Stdlib\CallbackHandler;
 
 /**
  * Class ApiResponse
  *
  * @package DvsaLogger\Listener
+ *
+ * @psalm-suppress PropertyNotSetInConstructor
```

Comment:
@psalm-suppress can be used either at function or line level. I don't believe it works at the class level
https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#docblock-suppression

Reply:
It's suppressing an error for $logger, which is set in the parent, so it is actually doing something. It's not suppressing for the whole class at once though, which is interesting.

Reply:
Yea I noticed this, I was hoping it might cover the class myself, but reading the documentation it looks like its only at the function level or the next line. Interesting that it's suppressing the error from it's parent.  I solved it by adding the offending member to the child class and either suppressed there or added typing.

Not sure what the best approach is with this.

Reply:
I've had a chat with @alexisc-kainos and we've gone with adding to the child and setting it in the constructor.

Reply:
Ahh OK, thanks for confirming that